.github/workflows/ci.yaml

```diff
@@ -167,20 +167,26 @@ jobs:
       - name: Run e2e tests
         run: make test-e2e
 
-  lint:
-    name: Lint
+  static-analysis:
+    name: Static analysis and linting
     runs-on: ubuntu-latest
-    timeout-minutes: 10
     steps:
       - uses: actions/checkout@v2
-      - name: Check linters are up to date
-        run: make check-update-linters
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.18'
+      - uses: newrelic/newrelic-infra-checkers@v1
+        with:
+          golangci-lint-config: golangci-lint
+      - name: Semgrep
+        uses: returntocorp/semgrep-action@v1
+        with:
+          auditOn: push
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.1.0
+        uses: golangci/golangci-lint-action@v3
+        continue-on-error: ${{ github.event_name != 'pull_request' }}
         with:
-          # Optional: show only new issues if it's a pull request. The default value is `false`.
           only-new-issues: true
```

Comment:
I feel like that bumping the version auto-enabled some additional linters and now we are taking more time. To solve the timeout issue in the ci we could try to add.
```
          args: --timeout 3m 
```

Possibly related to https://github.com/golangci/golangci-lint-action/issues/297

Reply:
It seems to work, thanks!